app/views/layouts/application.html.erb

```diff
@@ -6,17 +6,26 @@
     <title><%= t('appname') %></title>
     <%= csrf_meta_tags %>
 
-    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
-    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <% if (action_name === "new" && controller_name == "sessions" or controller_name === "users" && action_name == "new" or controller_name === "users" && action_name == "create") %>
+        <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track'=>'reload' %>
+        <%= stylesheet_link_tag 'navbar', media: 'all', 'data-turbolinks-track'=>'reload' %>
+    <% else %>
+        <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track'=>'reload' %>
+    <% end %>
+
+
+    <%= javascript_include_tag 'application', 'data-turbolinks-track'=> 'reload' %>
   </head>
 
-  <body>
+  <!--Display the background -->
+  <body class="<%= render partial: "application/background"%>">
```

Comment:
Just a thought: Wouldn't it be more elegant to put the computation of the background classes in a helper method inside of the application controller? Similar to the error_detector method.